### PR TITLE
Poison to Jason Migration

### PR DESCRIPTION
**Pull request by Waasi**
- Removed Poison and added Jason dependency
- Modified all adapters to use Jason instead of Poison
- Modified AdapterCase to use Jason instead of Poison as json decoder
- Modified Mailgun & Sparkpost test

### CODE REVIEW

**Comment by princemaple** [comment]
Hi @Waasi , thank you so much for picking up this one. PR looks good. One thing though, could poison be unlocked now? I don't think any of the dependencies directly depend on Poison? 👍 otherwise a great one, thanks again!

**Comment by Waasi** [comment]
Hi again @princemaple, It seems that inch_ex depends on Poison and Plug depends on Poison through inch_ex. Thanks again.

**Comment by princemaple** [comment]
Thank you for checking! I'll leave the merge to @stevedomin, as I don't have access to a work computer. Thanks again for the contribution! :blush:

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,4 @@
 erl_crash.dump
 *.ez
 .elixir_ls/
+*.swp
```
Comment (stevedomin) [inline .gitignore]: Can you remove this line please? Keen to keep this file Elixir/Erlang specific.

**Comment by yordis** [comment]
@stevedomin I would definitely make this a configuration because Jason has it own limitations (everything has to be Jason derive) for accomplish it performance.

Maybe we should add `config :swoosh, json_library: Jason` and fallback to Poison (because backwards compatibility) so every side is happy with it.

**Comment by princemaple** [comment]
Switching to Jason in Swoosh won't be a breaking change. Everyone who
depends on poison in their own code still gets to use poison.

**Comment by princemaple** [comment]
Thanks @Waasi !
I'll make the follow up change to make it configurable.

**Pull request merged**